test_pull_requests

```diff
@@ -152,10 +159,63 @@ def submitted_tests_for_branch(branch)
   $submitted_tests[branch] ? $submitted_tests[branch] : $submitted_tests['*']
 end
 
+# We do not expect the groups used to define trusted and administrative
+# members for this repo to change often, so we will 'cache' the group ID
+# to GitHub name and URL for them in a file on disk. The listing will
+# only be refreshed if the fils is deleted
+TEAM_CACHE_LOCATION = File.expand_path("~/.test_pull_requests_team_cache.json")
+Team_cache = if File.exists? TEAM_CACHE_LOCATION
+  JSON.parse(IO.read(TEAM_CACHE_LOCATION))
+else
+  {}
+end
+
 # Dynamically extend GitHubAPI to add methods that don't
 # require a local git project
 module Hub
   class GitHubAPI
+
+    # get_team_url_and_name returns the user-friendly URL and name for a GitHub team.
+    # If the team ID is found in the Team_cache, we will simply use it. If it
+    # is not, however, we will need to expend an API request to list all of the
+    # teams we care about and fill in the Team_cache before continuing.
+    def get_team_url_and_name(team_id)
+      $stderr.puts Team_cache
+      if Team_cache[team_id]
+        $stderr.puts "found cached value for #{team_id}: #{Team_cache[team_id]}"
+        return [Team_cache[team_id]['url'], Team_cache[team_id]['name']]
```

Comment:
Do you need to occasionally not use the cache.  Otherwise the data might get stale.